src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -55,13 +55,15 @@ public bool IsSuppressed (int id, MessageOrigin warningOrigin, out SuppressMessa
 			if (provider is not IMemberDefinition member)
 				return false;
 
-			MethodDefinition? userDefinedMethod = _context.CompilerGeneratedState.GetUserDefinedMethodForCompilerGeneratedMember (member);
-			if (userDefinedMethod == null)
-				return false;
-
-			Debug.Assert (userDefinedMethod != provider);
+			MethodDefinition? owningMethod;
+			while (_context.CompilerGeneratedState.TryGetOwningMethodForCompilerGeneratedMember (member, out owningMethod)) {
+				Debug.Assert (owningMethod != provider);
```

Comment:
Maybe this should be `owningMethod != member`, but it probably doesn't matter much.